collect_app/src/test/java/org/odk/collect/android/utilities/TextUtilsTest.java

```diff
@@ -0,0 +1,49 @@
+package org.odk.collect.android.utilities;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+
+@RunWith(RobolectricTestRunner.class)
+public class TextUtilsTest {
+
+    /**
+     * Should return null if provided with null and not throw a NPE.
+     *
+     * This is a silly test but it is here to guarantee this behaviour,
+     * since without it the method causes a crash when processing text for
+     * questions with no plain text label. See opendatakit/opendatakit#1247.
+     */
+    @Test
+    public void textToHtml_BouncesNullInput() {
+        String input = null;
+        CharSequence observed = TextUtils.textToHtml(input);
+        assertNull(observed);
+    }
+
+    /**
+     * Should convert Markdown to HTML.
+     *
+     * In the real app this call would produce the following string, but
+     * because the implementation uses android.Html.fromHtml, we end up with
+     * robolectric's implementation of fromHtml which isn't exactly the same.
+     * See opendatakit/collect#204.
```

Comment:
I think that my confusing wording and variables names made that unclear. Sorry. The output of `markdownToHtml` is the `<b>bold</b>` bit. `textToHtml` converts HTML tags to a styled `Spannable`, which when logged loses the styling. So I think that Robolectric is behaving the same way the Android platform does.

I agree however that this is really neither here nor there. I'd be fine removing this test altogether, leaving only the non-null test, which as you say was the point of the PR to begin with.